src/format/mod.rs

```diff
@@ -719,6 +796,81 @@ impl<'a, I: Iterator<Item = B> + Clone, B: Borrow<Item<'a>>> DelayedFormat<I> {
 impl<'a, I: Iterator<Item = B> + Clone, B: Borrow<Item<'a>>> fmt::Display for DelayedFormat<I> {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         format(f, self.date.as_ref(), self.time.as_ref(), self.off.as_ref(), self.items.clone())
+            .map_err(|_| fmt::Error)
+    }
+}
+
+/// A *temporary* object which can be used as an argument to `format!` or others.
+/// This is normally constructed via `format` methods of each date and time type.
+#[cfg(all(feature = "locales", any(feature = "alloc", feature = "std", test)))]
+#[derive(Debug)]
+pub struct DelayedFormatLocalized<'a, I> {
+    /// The date view, if any.
+    date: Option<NaiveDate>,
+    /// The time view, if any.
+    time: Option<NaiveTime>,
+    /// The name and local-to-UTC difference for the offset (timezone), if any.
+    off: Option<(String, FixedOffset)>,
+    /// An iterator returning formatting items.
+    items: I,
+    /// Locale used for text.
+    locale: &'a str,
```

Comment:
Something that might be really nice would be to create an `enum Locale` in `pure-rust-locales`, that would remove the lifetime here, it could also be the target of the `match` that I earlier said might be better in `pure-rust-locales`

Reply:
done